File: skeleton/src/main/java/me/shkschneider/skeleton/ui/transforms/ZoomOutTranformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class ZoomOutTranformer extends ABaseTransformer {

	@Override
	protected void onTransform(final View view, final float position) {
		final float scale = 1F + Math.abs(position);
		view.setScaleX(scale);
		view.setScaleY(scale);
		view.setPivotX(view.getWidth() * 0.5F);
		view.setPivotY(view.getHeight() * 0.5F);
		view.setAlpha(position < -1F || position > 1F ? 0F : 1F - (scale - 1F));
		if (position == -1) {
			view.setTranslationX(view.getWidth() * -1);
		}
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/ZoomOutTranformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View
import me.shkschneider.skeleton.extensions.then

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class ZoomOutTranformer : ABaseTransformer() {

    override fun onTransform(page: View, position: Float) {
        val scale = 1.toFloat() + Math.abs(position)
        page.scaleX = scale
        page.scaleY = scale
        page.pivotX = page.width * 0.1.toFloat()
        page.pivotY = page.height * 0.1.toFloat()
        page.alpha = (position < (-1).toFloat() || position > 1.toFloat()) then 1.toFloat() ?: 1.toFloat() - (scale - 1.toFloat())
        if (position == (-1).toFloat()) {
            page.translationX = (page.width * -1).toFloat()
        }
    }

}
